File: binV13/seventh_problem.js
const fs = require("fs");

const staffs = JSON.parse(fs.readFileSync("input_data/staff.json"));

const latestFeedbacks = staffs.map((staff) => {
  const latestFeedback = staff?.result?.feedback?.toSorted(
    (a, b) => new Date(b.date) - new Date(a.date),
  )[0];

  return {
    email: staff.email,
    full_name: staff.full_name,
    latest_feedback: latestFeedback?.comment ?? "No feedback available yet.",
  };
});

console.log(JSON.stringify({ data: latestFeedbacks }, null, 2));

File: binsV1/binV7/seventh_problem.rb
require 'json'
require 'date'

staffs = JSON.parse(File.read('input_data/staff.json'), symbolize_names: true)

latest_feedback = staffs.map do |staff|
  feedbacks = staff.dig(:result, :feedback).max_by do |feedback|
    Date.parse(feedback[:date])
  end

  {
    email: staff[:email],
    full_name: staff[:full_name],
    latest_comment: feedbacks&.fetch(:comment, 'No available feedback yet.')
  }
end

print JSON.pretty_generate({ data: latest_feedback })
